modules/blast/blast_primers/blast_primers.nf

```diff
@@ -0,0 +1,32 @@
+process blast_primers {
+    /*
+    Query primer FASTA files against annotated assembly using BLAST
+    */
+    tag "${sample}"
+
+    publishDir "${outdir}/${sample}/logs", mode: "${params.publish_mode}", overwrite: params.overwrite, pattern: "${task.process}/*"
+    publishDir "${outdir}/${sample}/blast", mode: "${params.publish_mode}", overwrite: params.overwrite, pattern: "primers/*.{json,json.gz}"
+
+    input:
+    set val(sample), file(blastdb) from BLAST_PRIMERS
+    file(query) from Channel.from(BLAST_PRIMER_FASTAS).collect()
+
+    output:
+    file("primers/*.{json,json.gz}")
+    file "${task.process}/*" optional true
+
+    when:
+    BLAST_PRIMER_FASTAS.isEmpty() == false
+
+    shell:
+    template(task.ext.template)
```

Comment:
For now we can replace this line with the actual code from the template.

This is how it would look

```nextflow
shell:
'''
CONTENTS_OF_THE_TEMPLATE

'''

```